client/my-sites/site-settings/date-time-format/date-time-format-options.jsx

```diff
@@ -0,0 +1,337 @@
+						<FormRadio
+							checked={ ! isCustomFormat && format === timeFormat }
+							disabled={ isRequestingSettings }
+							name="time_format"
+							onChange={ this.setTimeFormat }
+							value={ format }
+						/>
+						<span>{ now.format( phpToMomentDatetimeFormat( format ) ) }</span>
+					</FormLabel>
+				) }
+				<FormLabel className="date-time-format__custom-field">
+					<FormRadio
+						checked={ isCustomFormat }
+						disabled={ isRequestingSettings }
+						name="time_format"
+						onChange={ this.setCustomTimeFormat }
+						value={ timeFormat }
+					/>
+					<span>
+						{ translate( 'Custom', { comment: 'Custom date/time format field' } ) }
+						<FormInput
+							disabled={ isRequestingSettings }
+							name="time_format_custom"
+							onChange={ this.setCustomTimeFormat }
+							type="text"
+							value={ timeFormat || '' }
+						/>
+						<FormSettingExplanation>
+							{ isCustomFormat && timeFormat
+								? translate( 'Preview: %s', {
+									args: now.format( phpToMomentDatetimeFormat( timeFormat ) ),
+									comment: 'Date/time format preview'
+								} )
+								: ''
+							}
+						</FormSettingExplanation>
+					</span>
+					<FormSettingExplanation>
+						<ExternalLink
+							href="https://codex.wordpress.org/Formatting_Date_and_Time"
+							icon
+							target="_blank"
+						>
+							{ translate( 'Learn more about date and time formatting.' ) }
+						</ExternalLink>
+					</FormSettingExplanation>
+				</FormLabel>
+			</FormFieldset>
+		);
+	}
+
+	startOfWeekOption() {
+		const {
+			fields: { start_of_week: startOfWeek },
+			handleSelect,
+			isRequestingSettings,
+			translate,
+		} = this.props;
+
+		const daysOfWeek = [
+			translate( 'Sunday' ),
+			translate( 'Monday' ),
+			translate( 'Tuesday' ),
+			translate( 'Wednesday' ),
+			translate( 'Thursday' ),
+			translate( 'Friday' ),
+			translate( 'Saturday' ),
+		];
+
+		return (
+			<FormFieldset>
+				<FormLabel>
+					{ translate( 'Week starts on' ) }
+				</FormLabel>
+				<FormSelect
+					disabled={ isRequestingSettings }
+					name="start_of_week"
+					onChange={ handleSelect }
+					value={ startOfWeek || 0 }
+				>
+					{ daysOfWeek.map( ( day, index ) =>
+						<option key={ index } value={ index }>
+							{ day }
+						</option>
+					) }
+				</FormSelect>
+			</FormFieldset>
+		);
+	}
+
+	render() {
+		const {
+			handleSubmitForm,
+			isRequestingSettings,
+			isSavingSettings,
+			translate,
+		} = this.props;
+
+		return (
+			<div>
+				<SectionHeader>
+					<Button
+						compact={ true }
+						onClick={ handleSubmitForm }
+						primary={ true }
+						type="submit"
+						disabled={ isRequestingSettings || isSavingSettings }>
+							{ isSavingSettings
+								? translate( 'Saving…' )
+								: translate( 'Save Settings' )
+							}
+					</Button>
+				</SectionHeader>
+				<Card>
+					<form>
+						{ this.dateFormatOption() }
+						{ this.timeFormatOption() }
+						{ this.startOfWeekOption() }
```

Comment:
it's not that they are large, it's that they are _hidden_ from inside `render()` and then adding them into render could be unwieldy because of their size.

be wary of that existing structure (calling a bunch of functions inside `render()` to provide markup). we've been trying to discourage it for a long time in favor of more explicit code showing what's there or further modularization